internal/services/jsonrpc/openrpc/reflect.go

```diff
@@ -0,0 +1,267 @@
+package openrpc
+
+import (
+	"cmp"
+	"math/big"
+	"reflect"
+	"slices"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/kwilteam/kwil-db/core/types"
+	"github.com/kwilteam/kwil-db/core/types/decimal"
+)
+
+type MethodDefinition struct {
+	Description  string
+	Summary      string
+	RequestType  reflect.Type
+	ResponseType reflect.Type
+	RespTypeDesc string
+	// Errors ...
+}
+
+// InventoryAPI iterates through all handler types, making a map of all unique
+// OpenRPC schemas and building a []Method with every Param populated and
+// referencing a Schema if an "object" instead of a plain type.
+func InventoryAPI(handlerDefs map[string]*MethodDefinition, knownSchemas map[reflect.Type]Schema) []Method {
```

Comment:
Curious if using https://pkg.go.dev/go/ast could make this easier haha.

I'm not suggest change implementation, just feel like it's a perfect use case for go/ast'

Reply:
Not a bad point.  Establishing the hierarchy is what we're doing after all.  `go/ast` briefly occurred to me for the different job of getting method *descriptions* from the docs on types and struct fields too.
However, I wasn't sure about ast for this since afaict from the few uses I've seen of go/ast is that it has to build it from a source file.  Is there a way around that?  Looks like the strong suits of go/ast are for static code analysis and code generation, where as reflect is for dynamic (runtime) use such as when we register a bunch of concrete instantiations of services with a server.  Would love to learn more at some point, but I think it's working alright for now.